### DISCUSSION TITLE
Lesson 12, MintDsc Test

### DISCUSSION THREAD

**Discussion by Nureini** [Cyfrin/foundry-full-course-cu#321] [Q&A] [answered]
mintDsc is working here when testing even though the USER does not have 100 ether collateral deposited they only have 10.
What is the reason for this happening? Surely if this was on mainnet it would revert. 
``` solidity
    function testMintDscGivesUserTheCorrectAmountOfDscMinted() public {
        vm.startPrank(USER);
        ERC20Mock(weth).approve(address(dsce), 10 ether);
        dsce.depositCollateral(weth, 10 ether);
        dsce.mintDsc(100 ether);

        uint256 expectedTotalDscMinted = 100 ether;
        uint256 actualTotalDscMinted = dsce.getTotalDscMinted(USER);

        assertEq(expectedTotalDscMinted, actualTotalDscMinted);
        vm.stopPrank();
    }
```

**Comment by ghost** [comment, marked as answer]
Hello @Nureini, remember the health factor is calculated based on the usd value of the collateral. 

10 Ether * price of ether is a lot more than 100 DSC.

**Reply by Nureini** [reply]
thanks. i completely forgot that 1 DSC is pegged to 1 USD
